src/main/java/com/github/coleb1911/ghost2/commands/modules/fun/ModuleUrbanDictionary.java

```diff
@@ -0,0 +1,92 @@
+package com.github.coleb1911.ghost2.commands.modules.fun;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.github.coleb1911.ghost2.commands.meta.CommandContext;
+import com.github.coleb1911.ghost2.commands.meta.Module;
+import com.github.coleb1911.ghost2.commands.meta.ModuleInfo;
+import com.github.coleb1911.ghost2.utility.RestUtils;
+import discord4j.core.object.util.Permission;
+import discord4j.core.object.util.PermissionSet;
+import org.jsoup.Jsoup;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import javax.validation.constraints.NotNull;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.Objects;
+
+public class ModuleUrbanDictionary extends Module {
+
+    private final URI baseURI;
+    private final RestTemplate template;
+
+    public ModuleUrbanDictionary() {
+        this("https://mashape-community-urban-dictionary.p.rapidapi.com/", RestUtils.defaultRestTemplate());
+    }
+
+    public ModuleUrbanDictionary(String baseUrl, RestTemplate restTemplate) {
+        super(new ModuleInfo.Builder(ModuleUrbanDictionary.class)
+                .withName("urban")
+                .withDescription("Search random urban dictionary entry")
+                .withBotPermissions(PermissionSet.of(Permission.EMBED_LINKS)));
+
+        this.baseURI = URI.create(baseUrl);
+        this.template = restTemplate;
+    }
+
+    @Override
+    public void invoke(@NotNull CommandContext ctx) {
+
+        String searchString = String.join(" ", ctx.getArgs());
+        String urlSearchString = URLEncoder.encode(searchString, Charset.defaultCharset());
+        URI resultQuery = baseURI.resolve(baseURI.getPath() + "define?term=" + urlSearchString);
+        try {
+            ResponseEntity<UrbanDictionarySearchResults> results = template.getForEntity(resultQuery, UrbanDictionarySearchResults.class);
```

Comment:
You're not handling authentication with the API, unless I'm missing something. Invocation always results in a 401. Add an `X-RapidAPI-Key` header and a corresponding value to `GhostConfig`.

The absence of an API key should also be handled gracefully, rather than delegating it to the `RestClientException` catch block.